### PR TITLE
doc: add basic header docs

### PR DESCRIPTION
**Pull request by pbalcer**
This change is [Reviewable](https://reviewable.io/reviews/wlemkows/libuasync/5)

### CODE REVIEW

File: src/include/future.h [hunk 1]
```diff
@@ -1,5 +1,65 @@
 /* SPDX-License-Identifier: BSD-3-Clause */
-/* Copyright 2019-2021, Intel Corporation */
+/* Copyright 2021, Intel Corporation */
+
+/*
+ * future.h - oublic definitions for the future type, its associated state and
```
Comment (DamianDuy) [inline src/include/future.h:5]: It should be ```public```.

**Review by DamianDuy — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/wlemkows/libuasync/5#-:-MocEdMc40q2ugs_s8-o:bnfp4nl)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @DamianDuy and @pbalcer)

**Comment by kswiecicki** [comment]
___
*[src/future.c, line 51 at r2](https://reviewable.io/reviews/wlemkows/libuasync/5#-MondDaAEmgDst3CD0nD:-MondDaAEmgDst3CD0nE:b-10bs4g) ([raw file](https://github.com/wlemkows/libuasync/blob/beeef5acc4263db1156d2304f14dd172a580be41/src/future.c#L51)):*
<details><summary><i alt="summary">Quoted 12 lines of code…</i></summary>

> ```
> 			future_poll(&entry->future, waker);
> 			if (entry->future.context.state ==
> 				    FUTURE_STATE_COMPLETE &&
> 			    entry->map) {
> 				entry->map(&entry->future.context,
> 							next ?
> 							&next->future.context
> 							: ctx,
> 							entry->arg);
> 			} else {
> 				return FUTURE_STATE_RUNNING;
> 			}
> ```
</details>


The future that wasn't completed before and becomes complete after polling it (enters ```if (entry->future.context.state == FUTURE_STATE_COMPLETE && entry->map)```) won't cease the iteration to the next entry, as there's no ```break``` or ```return```. 
Is the future that becomes complete during ```async_chain_impl(struct future_context *ctx, struct future_waker waker)``` function execution considered as complete?

**Comment by kswiecicki** [comment]
:lgtm:

**Review by wlemkows — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/wlemkows/libuasync/5)* status: 0 of 4 files reviewed, 2 unresolved discussions (waiting on @kswiecicki and @pbalcer)

**Pull request merged**